Aydin/backend/src/lecturer/lecturer_controller.ts

```diff
@@ -0,0 +1,61 @@
+import {Router} from 'express';
+import {getManager, getConnection} from 'typeorm';
+import {Lecturer} from './lecturer_entity';
+import {Course} from '../course/course_entity';
+import {Student} from '../student/student_entity';
+import * as Joi from 'joi';
+import * as Validator from 'express-joi-validation';
+
+const validator = Validator.createValidator({});
+const bodySchema = Joi.object({
+  firstName: Joi.string().required(),
+  lastName: Joi.string().required(),
+  email: Joi.string().required(),
+  password: Joi.string().required(),
+});
+
+const LecturerController = Router();
+LecturerController.post('/', validator.body(bodySchema), async (req, res) => {
+  try {
+    const lecturer = new Lecturer();
+    lecturer.firstName = req.body.firstName;
+    lecturer.lastName = req.body.lastName;
+    lecturer.email = req.body.email;
+    lecturer.password = req.body.password;
+    const repo = getManager().getRepository(Lecturer);
+    await repo.save(lecturer);
+
+    res.sendStatus(201);
+  } catch (error) {
+    res.send(error);
+  }
+});
+LecturerController.put('/course', async (req, res) => {
+  try {
+    const courseRepo = getManager().getRepository(Course);
+    const studentRepo = getManager().getRepository(Student);
+    const course = await courseRepo.findOne({
+      where: {
+        CourseId: req.query.courseId,
+      },
+      relations: ['students'],
+    });
+    if (course === undefined) {
+      throw 'Course does not exists';
+    }
+    const student = await studentRepo.findOne({
+      where: {
+        studentId: req.body.studentId,
+      },
+    });
+    if (student === undefined) {
+      throw 'Student does not exists';
+    }
+    course.students = [student];
```

Comment:
why does `put` take exactly one student for the course and not multiple students? if this is just temporary code, please write a code comment

Reply:
it is not a temporary code, the lecturer can only add one student at a time, if that is the case what should i use instead of put?

Reply:
hm, why can the lecturer only add one student at a time? this seems like a weird requirement, in a real app I don't know if it would be like this. 


either way the code is confusing. does it always only set the course to be one student with [student]? it needs some comments

Reply:
yeah actually it was a bug and i fixed and i did a second pr, to be honest it didnt cross my mind to add multiple students at the same time, ill fix it so ill will add multiple students.